bika/lims/upgrade/v01_02_004.py

```diff
@@ -82,3 +88,13 @@ def remove_error_subfield_from_analysis_specs(portal, ut):
                 spec['warn_max'] = str(warn_max)
                 del spec['error']
         specs.setResultsRange(specs_rr)
+
+
+def reindex_reference_analysis(portal, ut):
+    # Update ReferenceAnalysis because new fields(Calculation & InterimFields)
+    # have been added on ReferenceAnalysis
+    catalog = api.get_tool('bika_analysis_catalog')
+    brains = catalog(portal_type='ReferenceAnalysis')
+    for brain in brains:
+        obj = brain.getObject()
+        obj.reindexObject()
```

Comment:
I strongly suggest to filter them by `review_state`. Imagine you have a database with more than 100k analyses.... this upgrade will take too looong. Notice, that you don't need to reindex those analyses that have already been submited, verified or published, so the total amount of analyses you'll need to reindex will, at a first glance, manageable.

Also, notice that reindexing analyses without assigning the Calculation and Interims will not have any effect.

Please, consider the following:

```python
exclude_states = ['to_be_verified', 'verified', 'published']
for brain in brains:
    if brain.review_state in exclude_states:
        continue
    analysis = api.get_object(brain)
    # Assign calculation and interims
    service = analysis.getAnalysisService()
    service = api.get_object(service)
    calc = service.getCalculation()
    if not calc:
        # No calculation, no need to reindex!
        continue
    analysis.setCalculation(calc)
    analysis.setInterimFields(calc.getInterimFields())
    analysis.setCalculation(calculation)
    analysis.reindexObject()
    logger.info("Updated Analysis '%s'  with calculation '%s'" % (analysis.Title(), calc.Title())
```